wlauto/instrumentation/trace_cmd/__init__.py

```diff
@@ -237,35 +241,49 @@ def update_result(self, context):  # NOQA pylint: disable=R0912
                 self.logger.debug(command)
-                process = subprocess.Popen(command, stderr=subprocess.PIPE, shell=True)
-                _, error = process.communicate()
-                if process.returncode:
-                    raise InstrumentError('trace-cmd returned non-zero exit code {}'.format(process.returncode))
-                if error:
-                    # logged at debug level, as trace-cmd always outputs some
-                    # errors that seem benign.
-                    self.logger.debug(error)
-                if os.path.isfile(local_txt_trace_file):
-                    context.add_iteration_artifact('txttrace', OUTPUT_TEXT_FILE, kind='export',
-                                                   description='trace-cmd generated ftrace dump.')
-                    self.logger.debug('Verifying traces.')
-                    with open(local_txt_trace_file) as fh:
-                        for line in fh:
-                            if 'EVENTS DROPPED' in line:
-                                self.logger.warning('Dropped events detected.')
-                                break
-                        else:
-                            self.logger.debug('Trace verified.')
+                if not os.path.isfile(local_trace_file):
+                    self.logger.warning('Not generating trace.txt, as {} does not exist.'.format(OUTPUT_TRACE_FILE))
                 else:
-                    self.logger.warning('Could not generate trace.txt.')
-            except OSError:
-                raise InstrumentError('Could not find trace-cmd. Please make sure it is installed and is in PATH.')
+                    try:
+                        process = subprocess.Popen(command, stderr=subprocess.PIPE, shell=True)
+                        _, error = process.communicate()
+                        if process.returncode:
+                            raise InstrumentError('trace-cmd returned non-zero exit code {}'.format(process.returncode))
+                        if error:
+                            # logged at debug level, as trace-cmd always outputs some
+                            # errors that seem benign.
+                            self.logger.debug(error)
+                    except OSError:
+                        raise InstrumentError('Could not find trace-cmd. Please make sure it is installed and is in PATH.')
+
+            else:
+                try:
+                    trace_file = self.output_file
+                    txt_trace_file = os.path.join(self.device.working_directory, OUTPUT_TEXT_FILE)
+                    command = 'trace-cmd report {} > {}'.format(trace_file, txt_trace_file)
+                    self.device.execute(command)
+                    self.device.pull_file(txt_trace_file, context.output_directory, timeout=pull_timeout)
+                except DeviceError:
+                    raise InstrumentError('trace-cmd returned non-zero exit code.')
```

Comment:
Error message should be updated to something like "Could not generate TXT report on target."